consonance-arch/src/main/java/info/consonance/arch/worker/WorkerRunnable.java

```diff
@@ -145,15 +144,41 @@ public void run() {
         int max = maxRuns;
 
         try {
+            // worker will need to pull from a specific queue
+            // for aws: curl http://169.254.169.254/latest/meta-data/instance-type
+            // for openstack: curl http://169.254.169.254/latest/meta-data/instance-type
+
             // the VM UUID
             log.info(" WORKER VM UUID provided as: '" + vmUuid + "'");
+
+            HttpClient client = new HttpClient();
+            // make really sure that we get a flavour
+            while (flavour == null) {
```

Comment:
Denis: ...because it might happen that the first request doesn't get a response if the metadata service is flaky, so need to keep trying until there is a flavour.